src/main/java/org/opensearch/knn/indices/Model.java

```diff
@@ -148,6 +129,7 @@ public boolean equals(Object obj) {
         Model other = (Model) obj;
 
         EqualsBuilder equalsBuilder = new EqualsBuilder();
+        equalsBuilder.append(getModelID(), other.getModelID());
         equalsBuilder.append(getModelMetadata(), other.getModelMetadata());
```

Comment:
As discussed before, i believe id is unique and hence comparing id on equals can be only condition for equals. What do you think?

Reply:
I think that there is a state of the cluster where 2 different models with the same ID are present:
1. Assume a model "test_model" is present in the cluster.
2. Then, a user submits another training request with model id "test_model". This model will be created [here](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/training/TrainingJob.java#L68-L79) and then validation will fail because model already exists [here](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/indices/ModelDao.java#L223). 

These models are not actually compared, but it seems that they are not equal and they could be present at the same time. Im still not sure if we should just simplify this to just comparing model id. Given that scenario, what do you think @VijayanB?

Reply:
I do see a case where 2 models with the same ID that are not equals are in the cluster at the same time:
1. A model "test_model" is created
2. A model "test_model" is submitted for training. The model will be created [here](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/training/TrainingJob.java#L68-L79) and rejected [here](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/indices/ModelDao.java#L223).

While they arent compared, it seems to me if they were this would be a problem. That being said, I am not quite sure how I feel about simplifying this. Given this case, what do you think @VijayanB ?

Reply:
if modelId already exists, we should not allow user to create a duplicate model? IMO, id should be unique not (modelid,status)

Reply:
Only the model object will be created. The model will not be persisted. The request will fail [here](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/training/TrainingJobRunner.java#L96-L117).

Adding duplication check logic in the Model class's constructor would be very complex given the multinode and multithreaded nature of OpenSearch.

That being said, I guess I cannot think of a case where 2 different valid models with the same ID would be present on a node. I will go ahead and simplify this logic.